app/assets/javascripts/components/story/CollapsedStory/index.js

```diff
@@ -23,39 +27,177 @@ const storyClassName = (story, additionalClassname = '') => {
       'Story--estimated': !isStoryNotEstimated
     },
     additionalClassname
-  );
-};
-
-export const CollapsedStory = ({ onToggle, story, updateCollapsedStory, project, className, title }) =>
-  <div
-    className={storyClassName(story, className)}
-    onClick={onToggle}
-    title={title}
-  >
-    <StoryPopover story={story}>
-      <div className='Story__icons-block'>
-        <StoryIcon storyType={story.storyType} />
-        <CollapsedStoryEstimate estimate={story.estimate} />
-        <StoryDescriptionIcon description={story.description} />
-      </div>
-    </StoryPopover>
-
-    <CollapsedStoryInfo story={story} />
-
-    <CollapsedStoryStateActions story={story}
-      onUpdate={(newAttributes) =>
-        updateCollapsedStory(story.id, project.id, newAttributes)}
-    />
-  </div>
+  )
+}
+
+const type = {
+  story: 'STORY'
+}
+
+const calculatePosition = (aboveStory, bellowStory, storiesArray, index) => {
+  if (bellowStory === undefined) {
+    return (Number(storiesArray[index].position) + 1)
+  }else if (aboveStory !== undefined && bellowStory !== undefined) {
+    return (Number(bellowStory.position) + Number(aboveStory.position)) / 2;
+  }else if (aboveStory === undefined) {
+    return (Number(storiesArray[index].position) - 1);
+  }
+}
+
+const getNewPosition = (item, story, storiesArray, index) => {
+  let aboveStory;
+  let bellowStory;
+
+  if(item.index > index) {
+    aboveStory = storiesArray[index - 1]
+    bellowStory = story
+  } else {
+    bellowStory = storiesArray[index + 1]
+    aboveStory = story
+  }
+
+  return calculatePosition(aboveStory, bellowStory, storiesArray, index)
+}
+
+const checkStoryPosition = (ref, dragIndex, hoverIndex, monitor) => {
+  const hoverBoundingRect = ref.current.getBoundingClientRect()
+  const hoverMiddleY = (hoverBoundingRect.bottom - hoverBoundingRect.top) / 2
+  const clientOffset = monitor.getClientOffset()
+  const hoverClientY = clientOffset.y - hoverBoundingRect.top
+  
+  if(dragIndex === hoverIndex) {
+    return true
+  }
+
+  if (dragIndex < hoverIndex && hoverClientY < hoverMiddleY) {
+    return true
+  }
+
+  if (dragIndex > hoverIndex && hoverClientY > hoverMiddleY) {
+    return true
+  }
+}
```

Comment:
There is a chance to return nothing in this function.

Also, you could extract these tree `if`s into functions, for example:
```js
// Choose a better name for these function :D
const isDraggedInTheSameIndex = (dragIndex, hoverIndex) => dragIndex === hoverIndex
const isDragged1 = (dragIndex, hoverIndex, hoverClientY, hoverMiddleY) => dragIndex < hoverIndex && hoverClientY < hoverMiddleY
const isDragged2 = (dragIndex, hoverIndex, hoverClientY, hoverMiddleY) => dragIndex > hoverIndex && hoverClientY > hoverMiddleY
const isDraggedInside = (dragIndex, hoverIndex, hoverClientY, hoverMiddleY) => 
  isDragged1(dragIndex, hoverIndex, hoverClientY, hoverMiddleY) || isDragged2(dragIndex, hoverIndex, hoverClientY, hoverMiddleY)
// Choose a better name for these function :D

const checkStoryPosition = (ref, dragIndex, hoverIndex, monitor) => {
  const hoverBoundingRect = ref.current.getBoundingClientRect()
  const hoverMiddleY = (hoverBoundingRect.bottom - hoverBoundingRect.top) / 2
  const clientOffset = monitor.getClientOffset()
  const hoverClientY = clientOffset.y - hoverBoundingRect.top
  
  return isDraggedInTheSameIndex(dragIndex, hoverIndex) 
    || isDraggedInside(dragIndex, hoverIndex, hoverClientY, hoverMiddleY)
}
```